### ISSUE TITLE
collaboration diagram should include smart pointer members

### ISSUE THREAD

**Issue by jwuttke** [doxygen/doxygen#8948]
source.h:
```
#include <memory>
class Bar {};
class Baz {};
class Foo {
public:
  Bar member1;
  std::unique_ptr<Baz> member2;
};
```
Running  `doxygen -g; doxygen` yields `html/classFoo.html` that looks like

![result](https://user-images.githubusercontent.com/5262000/145024801-89dd2b1c-6153-4d99-9eae-5c8b1f4fe677.png)

It would be great if the collaboration diagram of `Foo` could point not only to `Bar`, but also to `Baz`.

Ideally, the arrow should be plotted in different colors for plain members, unique_ptr, and shared_ptr.

Full example: 
[example.zip](https://github.com/doxygen/doxygen/files/7668171/example.zip)

**Comment by albert-github** [comment]
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by jwuttke** [comment]
doxygen -v => 1.9.1

**Comment by jwuttke** [comment]
@albert-github : changed issue text to show full input, and the relevant part of the output.

**Comment by albert-github** [comment]
I assume you also  set `HAVE_DOT` to `YES`, therefor always:
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be **persistent.**

**Comment by jwuttke** [comment]
ok, zip attached. tar upload is not possible through this web interface.

**Comment by albert-github** [comment]
I normally have no problems wit compressed tar files )like: `tar czf example.tar.gz <files>`), except in case there are `.bat` or `.js` etc. files in it (but the same applies for `.zip` compression).

**Comment by jwuttke** [comment]
I see, extension `.tgz` is not supported, but `.gz` is.

**Comment by albert-github** [comment]
Regarding the `HAVE_DOT` "confusion" I see that in your "standard Doxyfile" you have:
```
# If you set the HAVE_DOT tag to YES then doxygen will assume the dot tool is
# available from the path. This tool is part of Graphviz (see:
# http://www.graphviz.org/), a graph visualization toolkit from AT&T and Lucent
# Bell Labs. The other options in this section have no effect if this option is
# set to NO
# The default value is: YES.

HAVE_DOT               = YES
```

Whilst in the official Doxyfile gives:
```
# If you set the HAVE_DOT tag to YES then doxygen will assume the dot tool is
# available from the path. This tool is part of Graphviz (see:
# http://www.graphviz.org/), a graph visualization toolkit from AT&T and Lucent
# Bell Labs. The other options in this section have no effect if this option is
# set to NO
# The default value is: NO.

HAVE_DOT               = NO
```

Probably the distributor if doxygen for your platform "fixed" this (I think I saw something like this happen in the past).

**Comment by doxygen** [comment]
Looks like this still worked in version 1.9.0. At least when `BUILTIN_STL_SUPPORT = YES`

**Comment by jwuttke** [comment]
Indeed, `BUILTIN_STL_SUPPORT = YES` soves my problem. Many thanks!

**Comment by albert-github** [comment]
Problem is that in the 1.9.2 version the problem reappears even when `BUILTIN_STL_SUPPORT = YES` so we have here to do with a regression in newer versions. Reopening

**Comment by jwuttke** [comment]
Also, the result under 1.9.1 is not ideal.

Slightly extended example:
```#include <memory>
class Arx;
class Bar : public Arx {};
class Baz : public Arx {};
class Foo {
public:
  Bar member1;
  std::unique_ptr<Baz> member2;
};
```
yields
![result2](https://user-images.githubusercontent.com/5262000/145043214-7d084977-7a7b-487b-ad4c-c14ec07e4820.png)

Ideally, the box would not show `std::unique_ptr<Baz>` but just `Baz`, and the fact that membership in `Foo` is mediated by a unique_ptr would be indicated differently, e.g. by the color of the arrow or by a suffix to the label `member2`.

And for sure, we should see that `Baz` inherits from `Arx`.

**Comment by doxygen** [comment]
@jwuttke In your example `Arx` is just a forward declaration, not a definition, so not something doxygen can link to.

**Comment by jwuttke** [comment]
Example corrected, result still the same.

**Comment by albert-github** [comment]
@doxygen are you looking into the problem already (I could start a bisect otherwise).

**Comment by albert-github** [comment]
In the mean time I did a bisect:
```
37deea7dc7f4d3e6d1df167471a01d3933a65602 is the first bad commit
commit 37deea7dc7f4d3e6d1df167471a01d3933a65602
Date:   Mon Feb 8 21:21:26 2021 +0100

    Refactoring: replace QRegExp by std::regex in util.cpp
```

**Comment by doxygen** [comment]
@albert-github thanks, I'll have a look what happened

**Comment by doxygen** [comment]
The referenced commit should fix the problem introduced in 1.9.2, but there is still a regression between 1.9.0 and 1.9.1 that needs further investigation.

**Comment by doxygen** [comment]
With the latest commit, the behavior of version 1.9.0 should be restored.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
